Source/SwiftyDropbox/Platform/SwiftyDropbox_iOS/OAuthMobile.swift

```diff
@@ -68,15 +68,13 @@ open class DropboxMobileOAuthManager: DropboxOAuthManager {
         super.init(appKey: appKey, host:host)
         self.urls.append(self.dauthRedirectURL)
     }
-    
-    internal override func extractFromUrl(_ url: URL) -> DropboxOAuthResult {
-        let result: DropboxOAuthResult
+
+    internal override func extractFromUrl(_ url: URL, completion: @escaping DropboxOAuthCompletion) {
         if url.host == "1" { // dauth
-            result = extractfromDAuthURL(url)
+            extractfromDAuthURL(url, completion: completion)
```

Comment:
the code flow is async, because after extract the data from URL, the SDK has to make an api request to get the actual access token back.
it is `handleCodeFlowResults()` in `extractfromDAuthURL()` that's async.